portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,68 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
+import java.util.ArrayList;
+import com.google.gson.Gson;
+ 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  
+  private ArrayList<String> names = new ArrayList<String>();
+ 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query("Task").addSort("comment", SortDirection.DESCENDING);
+ 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+ 
+    ArrayList<String> commentsFromDB = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+        String entityComment = (String) entity.getProperty("comment");
+        commentsFromDB.add(entityComment);
+    }
+ 
+    String json = convertToJsonUsingGson(commentsFromDB);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String inputText = getParameter(request, "text-input", "");
+    names.add(inputText);
+
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("comment", inputText);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+
+    response.sendRedirect("/index.html");
```

Comment:
Not required change. Just some info for you.

I remember you said the POST was causing a page reload. It was actually because of this line. If you don't want that behavior, you should update this to send JSON down like the GET.

```
response.setContentType("application/json;");
response.getWriter().println(json);
```